### PR TITLE
use default shasum for os x if present

### LINKED ISSUE #450: lib: newlib and libc++ scripts assume sha256sum
We use SHA256 to verify the download, but we assume the user has a `sha256sum` binary. This isn't the case on mac by default. We should handle this case somehow.

### PR DESCRIPTION
**Pull request by ppannuto**
Fixes #450.

### CODE REVIEW

File: lib/fetch-libc++.sh [hunk 1]
```diff
@@ -19,14 +19,21 @@ MIRRORS=(\
...
+
 let FOUND=0
 
 # Try from each mirror until we successfully download a .zip file.
 for MIRROR in ${MIRRORS[@]}; do
   URL=$MIRROR/$ZIP_FILE
   echo "Fetching libc++ from ${MIRROR}..."
   echo "  Fetching ${URL}..."
-  wget -O $ZIP_FILE  "$URL" && (echo "$GCC_SHA $ZIP_FILE" | sha256sum -c)
+  # Note: There must be two space characters for `shasum` (sha256sum doesn't care)
```
Comment (alevy) [inline lib/fetch-libc++.sh:35]: Two space characters where? what is this referring to?

File: lib/fetch-libc++.sh [hunk 2]
```diff
@@ -19,14 +19,21 @@ MIRRORS=(\
   "https://alpha.mirror.svc.schuermann.io/files/tock"\
 )
 
+if test -x /usr/bin/shasum; then
```
Comment (alevy) [inline lib/fetch-libc++.sh:22]: This _at least_ needs a comment. Why this particular path? Do we always prefer `shasum` if it is present? Or only if `sha256sum` _isn't_ present?

Is this the `shasum` that comes with Perl? If so, Perl is a pre-requisite for most Linux distributions in practice, does it make sense to just use `shasum`?

File: lib/fetch-newlib.sh [hunk 3]
```diff
@@ -17,14 +17,21 @@ MIRRORS=(\
   "https://alpha.mirror.svc.schuermann.io/files/tock"\
 )
 
+if test -x /usr/bin/shasum; then
```
Comment (alevy) [inline lib/fetch-newlib.sh:20]: That this is duplicated is a bit of a red flag.

**Review by alevy — CHANGES_REQUESTED** [review]
I think this is hacks on hacks that is going to be impossible to reliably test and maintain.

File: lib/fetch-libc++.sh (hunk 1 above)
Comment (bradjc) [reply lib/fetch-libc++.sh:35]: Between the correct sha and the file name:

```
❯ echo "1e6f33cd5f44acefebc1ef677db4be33829a5228  README.md" | shasum -a 256 -c
README.md: OK
❯ echo "1e6f33cd5f44acefebc1ef677db4be33829a5228 README.md" | shasum -a 256 -c
shasum: standard input: no properly formatted SHA checksum lines found
```

File: lib/fetch-libc++.sh (hunk 2 above)
Comment (bradjc) [reply lib/fetch-libc++.sh:22]: This is what happened in the kernel: https://github.com/tock/tock/pull/1669. Should we have done something differently? I think we just want this to work on as many systems as possible.

**Review by lschuermann — APPROVED** [review]
Solution looks good to me, and this does solve a real problem.

It's disappointing that this is still an issue, and (apart from all the other reasons why this is bad) we probably don't want to require a host C/Rust toolchain to build our own sha256sum tool in this repo.

**Comment by ppannuto** [comment]
> I think this is hacks on hacks that is going to be impossible to reliably test and maintain.

For better or worse, it's just propagating over the existing approach (hack? :) ) from the kernel repo, which has been unchanged for years now. Think this hits least-bad pragmatic approach for the current ecosystem of developer machines writ large.

**Review by alevy — APPROVED** [review]

**Pull request merged**